ci/install.sh

```diff
@@ -1,3 +1,24 @@
+
+# Optionally, install development versions of dependenies
+if [[ ${UPSTREAM_DEV} ]]; then
+    # FIXME https://github.com/mamba-org/mamba/issues/412
+    # mamba uninstall --force numpy pandas scikit-learn
+    conda uninstall --force numpy pandas scikit-learn
+
+    python -m pip install --no-deps --pre \
+        -i https://pypi.anaconda.org/scipy-wheels-nightly/simple \
+        numpy \
+        pandas \
+        scikit-learn
+
+    python -m pip install \
+        --upgrade \
+        git+https://github.com/pydata/sparse \
+        git+https://github.com/dask/dask \
+        git+https://github.com/dask/distributed
```

Comment:
Looking at the output from this step in CI, the development version of `numpy` is getting installed above, but then overwritten to `numpy=1.20.3` here for some reason (see [here](https://github.com/dask/dask-ml/runs/4217350233?check_suite_focus=true#step:4:182)). All the other upstream dependencies are installed as expected

Reply:
Opted to go with `dask`, `distributed`, and `scikit-learn` as the upstream packages to test against